smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -0,0 +1,224 @@
+        }
+    }
+
+    /**
+     * NoninclusiveTermsValidator configuration.
+     */
+    public static final class Config {
+        private Map<String, List<String>> noninclusiveTerms = MapUtils.of();
+        private boolean appendDefaults = false;
+
+        public Map<String, List<String>> getNonInclusiveTerms() {
+            return noninclusiveTerms;
+        }
+
+        public void setNoninclusiveTerms(Map<String, List<String>> terms) {
+            this.noninclusiveTerms = terms;
+        }
+
+        public boolean isAppendDefaults() {
+            return appendDefaults;
+        }
+
+        public void setAppendDefaults(boolean appendDefaults) {
+            this.appendDefaults = appendDefaults;
+        }
+    }
+
+    final Map<String, List<String>> termsMap;
+
+    private NoninclusiveTermsValidator(Config config) {
+        if (config.isAppendDefaults()) {
+            termsMap = new HashMap<>(BUILT_IN_NONINCLUSIVE_TERMS);
+            termsMap.putAll(config.getNonInclusiveTerms());
+        } else if (!config.getNonInclusiveTerms().isEmpty()) {
+            termsMap = MapUtils.copyOf(config.getNonInclusiveTerms());
+        } else {
+            termsMap = new HashMap<>(BUILT_IN_NONINCLUSIVE_TERMS);
+        }
+    }
+
+    /**
+     * Runs a full text scan on a given model and stores the resulting TextOccurrences objects.
+     *
+     * Namespaces are checked against a global set per model.
+     *
+     * @param model Model to validate.
+     * @return a list of ValidationEvents found by the implementer of getValidationEvents per the
+     *          TextOccurrences provided by this traversal.
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        TextIndex textIndex = TextIndex.of(model);
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        for (TextInstance text : textIndex.getTextInstances()) {
+            getValidationEvents(text, validationEvent -> {
+                validationEvents.add(validationEvent);
+            });
+        }
+        return validationEvents;
+    }
+
+    /**
+     * This is a somewhat abstracted interface for validating text instances in a model
+     * by pushing ValidationEvents into a provided consumer.
+*
+     * @param occurrence text occurrence found in the body of the model
+     * @param validationEventConsumer consumer to push ValidationEvents into
+     */
+    private void getValidationEvents(TextInstance instance,
+                                       Consumer<ValidationEvent> validationEventConsumer) {
+        for (Map.Entry<String, List<String>> termEntry : termsMap.entrySet()) {
+            //lower casing the term will be more necessary when the terms are from config
+            if (containsTerm(instance.getText(), termEntry.getKey())) {
+                switch (instance.getLocationType()) {
+                    case NAMESPACE:
+                        validationEventConsumer.accept(ValidationEvent.builder()
```

Comment:
If this can't just use `warning` (since a namespace doesn't have a singular Shape to pass to it?) it's probably worth documenting.